### PR TITLE
Add support for registerDna

### PR DESCRIPTION
**Pull request by zippy**
Updates to holochain-conductor api dev16 and adds support for registerDna.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,8 +95,15 @@ orchestrator.registerScenario('proper zome call', async (s, t) => {
...
-  const carol_test_happ_with_bobs_test_key = await carol.installHapp([dnaTest], bob_blog_happ.agent)
+
+  // or a happ with a previously generated key and previously registered dna by hash
+  // which you can get from the cells as returned in the InstalledHapp from previous install
+  const testDnaHash = bob_test_happ.cells[0].dnaHash()
+  const carol_test_happ_with_bobs_test_key = await carol.installHapp([testDnaHash], bob_blog_happ.agent)
+
+  // or from a call to registerDna which might look like this to create a derived dna using
+  // different uuid.
+  const derivedDnaHash = await carol.registerDna({hash: testDnaHash}, "1234567890")
```
Comment (timotree3) [inline README.md:106]: I don't fully understand. Does this mean that if you install a DNA with one conductor, then another conductor on the same machine has access to that DNA and will attempt to look for it whenever it receives the DNA hash? I was assuming that each conductor running on the same machine has entirely separate databases.

File: README.md [hunk 2]
```diff
@@ -95,8 +95,15 @@ orchestrator.registerScenario('proper zome call', async (s, t) => {
 
   // and install a single happ
   const carol_blog_happ = await carol.installHapp([dnaBlog])
-  // or a happ with a previously generated key
-  const carol_test_happ_with_bobs_test_key = await carol.installHapp([dnaTest], bob_blog_happ.agent)
+
+  // or a happ with a previously generated key and previously registered dna by hash
+  // which you can get from the cells as returned in the InstalledHapp from previous install
```
Comment (timotree3) [inline README.md:100]: I find this sentence hard to parse.
```
  // or a happ with a previously generated key and previously registered dna by hash
  // which you can get from the cells as returned in the InstalledHapp from previous install
```
Maybe I can rewrite it once I understand how the feature works...

**Review by timotree3 — APPROVED** [review]
The code changes look good to me but I don't understand how this feature works.

File: test/e2e/test-dynamic-on.ts [hunk 3]
```diff
@@ -74,7 +74,25 @@ module.exports = (testOrchestrator, testConfig) => {
     })
 
     const stats = await orchestrator.run()
+    t.equal(stats.successes, 1)
+    t.end()
+  })
 
+  test('dna registration', async t => {
```
Comment (zo-el) [inline test/e2e/test-dynamic-on.ts:81]: Can we check that one dna is initially installed and then after the `registerDna()`  it should return 2 dnas

File: test/e2e/test-dynamic-on.ts [hunk 4]
```diff
@@ -74,7 +74,25 @@ module.exports = (testOrchestrator, testConfig) => {
...
+    const [conductorConfig, installApps] = testConfig()
+    const orchestrator = testOrchestrator()
+    orchestrator.registerScenario('we can register Dnas', async s => {
+      const [ alice ] = await s.players([conductorConfig])
+      const [[alice_happ]] = await alice.installAgentsHapps(installApps)
+
```
Comment (zo-el) [inline test/e2e/test-dynamic-on.ts:87]:
```suggestion
      let dnas =  await alice.adminWs().listDnas()
      t.equal(dnas.length, 1)
```

File: test/e2e/test-dynamic-on.ts [hunk 5]
```diff
@@ -74,7 +74,25 @@ module.exports = (testOrchestrator, testConfig) => {
...
+      const dnaHash = alice_happ.cells[0].dnaHash()
+      const derivedDnaHash = await alice.registerDna({hash: dnaHash}, "12345")
+      t.equal(dnaHash.length, 39)
+      t.equal(derivedDnaHash.length, 39)
+      t.notEqual(dnaHash, derivedDnaHash)
+    })
```
Comment (zo-el) [inline test/e2e/test-dynamic-on.ts:93]:
```suggestion
      dnas =  await alice.adminWs().listDnas()
      t.equal(dnas.length, 2)

    })
```

**Review by zo-el — APPROVED** [review]
Looks good. Just one suggestion to update the test.

File: README.md (hunk 1 above)
Comment (zippy) [reply README.md:106]: oops, bad example.  They are separate.  I'll fix.

File: README.md [hunk 6]
```diff
@@ -95,8 +95,14 @@ orchestrator.registerScenario('proper zome call', async (s, t) => {
...
+  // or from a call to registerDna which might look like this to create a derived dna using
+  // a different uuid
+  const blogDnaHash = carol_test_happ.cells[0].dnaHash()
+  const derivedDnaHash = await carol.registerDna({hash: blogDnaHash}, "1234567890")
```
Comment (timotree3) [inline README.md:104]: Should this say `testDnaHash`?

File: README.md [hunk 7]
```diff
@@ -95,8 +95,14 @@ orchestrator.registerScenario('proper zome call', async (s, t) => {
 
   // and install a single happ
   const carol_blog_happ = await carol.installHapp([dnaBlog])
-  // or a happ with a previously generated key
-  const carol_test_happ_with_bobs_test_key = await carol.installHapp([dnaTest], bob_blog_happ.agent)
+
+  // or a happ with a previously generated key and previously registered dna by hash
+  // which you can get from the cells as returned in the InstalledHapp from previous install
+  // or from a call to registerDna which might look like this to create a derived dna using
+  // a different uuid
```
Comment (timotree3) [inline README.md:102]:
```suggestion
  // or install a happ using
  // - a previously generated key
  // - and the hash of a dna that was previously registered with the same conductor
  // (a dna can be registered either by installing a happ with that dna or by calling registerDna with an old dna's hash and a new UUID)
```

File: README.md (hunk 6 above)
Comment (zippy) [reply README.md:104]: In the example carol has a blogDna installed.  She then uses creates a derived DNA from that hash and registers a new derived dna.
